department-of-veterans-affairs/vets-website
# Uae/90974/add api call and mock for claim status

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary

- Adds `claims` to include parameter on api calls to get past appointments
- Adds `claims` to include parameter on api calls to get a past appointment
- Adds feature flag
- Adds unit tests for past appointments list and appointment details view
- Modifies confirmed appointments mocks to include claim data for testing and future development on the front-end

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
[department-of-veterans-affairs/va.gov-team#90974](https://github.com/department-of-veterans-affairs/va.gov-team/issues/90974)

## Testing done

- Unit tests that test that the `claims` string only gets added on past appointments and when the feature toggle is enabled
- Navigate to the appointments list view and see that the claims include is added to the call only on past appointments
- Navigate to a past appointment and refresh and see that the claims include is added to the api call for a single appointment
- Navigate to current appointments and see that claims is not added
- Navigate to a current appointment details view and on refresh see that claims is not added 

## Screenshots

## What areas of the site does it impact?

Vaos/appointments app

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
